### PR TITLE
Feature/date switcher

### LINKED ISSUE #384: Date switcher
In date dependent views it should be possible to jump to a given date using a calendar widget that drops down clicking on the current date.

### PR DESCRIPTION
**Pull request by fadwamahmoud**
Fixes #384 
- Calendar icon is clicked to show the date picker.
- On slots view only Mondays are clickable.

### CODE REVIEW

File: eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from "react";
+import { DateTime, DateTimeUnit } from "luxon";
+import { useDispatch } from "react-redux";
+
+import { CalendarPicker } from "@mui/lab";
+
+import { changeCalendarDate } from "@/store/actions/appActions";
+import Menu from "@mui/material/Menu";
```
Comment (ikusteu) [inline eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx:8]: Here you can import `MenuProps` (as a named import)...Mui luckily provides the props interface for each of their components right by the component file...like we do in each `<componentFolder>/index.ts`

File: eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+interface Props {
+  currentDate: DateTime;
+  /**
+   * Value for next/prev timeframe start time calculation
+   */
+  jump?: DateTimeUnit;
+  anchorEl: null | HTMLElement;
+  open: boolean;
+  handleClose: () => void;
```
Comment (ikusteu) [inline eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx:18]: Here you can simply extend the aforementioned `MenuProps`:
```typescript
interface Props extends MenuProps {
  jump: DateTimeUnit // good one btw ;)
  currentDate: DateTime
}
```

Comment (ikusteu) [reply eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx:18]: This way the `onClose`, `open` and `anchorEl` will be included in the props interface. About `onClose` / `handleClose`, it is a convention to use `handle<something>` (i.e. `handleClick`) when you're writing a handler within the component, while all handlers received as props should be `on<something>` (i.e. `onClick`)

Example:
```typescript
const Button = ({ onClick, children }) => <button {...{ onClick }} >{children}</button>

const ButtonBox = () => {
  const handleClick = () => {/* do something */}
  return <Button onClick={handleClick} >Click Me</Button> 
}
```

File: eisbuk-admin/src/components/atoms/DateSwitcher/__tests__/DateSwitcher.test.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        <LocalizationProvider dateAdapter={AdapterDateFns}>
+          <DateSwitcher
+            currentDate={testDateDateTime}
+            open={true}
+            anchorEl={null}
+            handleClose={() => {}}
+          />
+        </LocalizationProvider>
+      );
+    });
```
Comment (ikusteu) [inline eisbuk-admin/src/components/atoms/DateSwitcher/__tests__/DateSwitcher.test.tsx:38]: No need for smoke test, after you've written the stuff down below...successful rendering is implied for all of the other tests

File: eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      <Menu anchorEl={anchorEl} open={open} onClose={handleClose}>
+        <CalendarPicker
+          shouldDisableDate={disableNonMondays}
+          date={currentDate}
+          onChange={(currentDate) => {
+            dispatch(changeCalendarDate(currentDate!));
+          }}
+        />
+      </Menu>
+    </>
```
Comment (ikusteu) [inline eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx:44]: The menu doesn't close on outside click (and can't be closed at all)

**Review by ikusteu — CHANGES_REQUESTED** [review]

File: eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx (hunk 4 above)
Comment (ikusteu) [reply eisbuk-admin/src/components/atoms/DateSwitcher/DateSwitcher.tsx:44]: My bad it most definitely works

**Review by silviot — APPROVED** [review]
Nice! It's exactly how I imagined it should be. I think it improves usability quite a bit. Good job!

**Review by ikusteu — APPROVED** [review]

**Pull request merged**